mods/beds/api.lua

```diff
@@ -97,6 +99,20 @@ function beds.register_bed(name, def)
 			type = "fixed",
 			fixed = {0, 0, 0, 0, 0, 0},
 		},
+		on_destruct = function(pos)
```

Comment:
l added the on_destruct to only one part of the bed.
https://github.com/minetest/minetest_game/pull/488/files#diff-b2333dc7fcdfecb27e6bed8daa7d3af8L40
Maybe it's faster if you use a local variable instead of the meta.